### PR TITLE
Searching / Private Messaging / Requests (REFACTOR) / Categories / Single Post URI / Public (LOCAL / REMOTE) / Sharing Feature

### PR DESCRIPTION
**Pull request by kezzayuno**
# Description

Essentially how this works is that the postFrom attribute associated with a post is who made the private post. The post itself is sent to the receiver Author's inbox and the created post is sent to the post db of the Author that wrote the post. 

Frontend Messages essentially requests all the private posts -> filters by postFrom given the currentMessenger they are on (in the future, indicated by the left nav ("highlighted button or something")). When they click on a different name on the left nav, a new set of posts will show from that author (who was clicked); the left nav shows all the authors who have privately sent posts to the viewer author (author who is currently logged in looking at the private messages) 

I am not sure entirely how this will work with remote; I will explain to them later how to send private messages to us; essentially they just have to send an extra attribute in their post called postFrom: <AuthorId who wrote the post> and have the URL /authors/:authorId who receives the post/inbox (POST request).  

TLDR; ReceiverAuthor gets SenderAuthor's private post and SenderAuthor saves that post in their post db

Hopefully this explains it, this was a bit tough to code.

### CODE REVIEW

**Comment by kezzayuno** [comment]
Still need to test Requests refactor
